runtime/lua/vim/net.lua

```diff
@@ -25,7 +25,7 @@ function M.request(url, opts, on_response)
   local retry = opts.retry or 3
 
   -- Build curl command
-  local args = { 'curl' }
+  local args = { 'curl', '--max-time', '5' }
```

Comment:
Not the final solution, but need to find a way to have this behaviour. Either by allowing curl arguments to `vim.net.request` or some callback-wizardry.

Reply:
#36164 Adds a `:cancel()` method, thus this should not be necessary anymore (the caller can handle the timeout). Until that is merged, this PR just prints a warning and doesn't cancel the tasks.